main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidatorTest.java

```diff
@@ -17,110 +17,356 @@
+          ARRIVAL_TIME_FIELD_NAME,
+          DEPARTURE_TIME_FIELD_NAME,
+          STOP_ID_FIELD_NAME,
+          STOP_SEQUENCE_FIELD_NAME,
+          STOP_HEADSIGN_FIELD_NAME,
+          PICKUP_TYPE_FIELD_NAME,
+          DROP_OFF_TYPE_FIELD_NAME,
+          CONTINUOUS_PICKUP_FIELD_NAME,
+          CONTINUOUS_DROP_OFF_FIELD_NAME,
+          SHAPE_DIST_TRAVELED_FIELD_NAME
+        });
+  }
+
+  private static CsvHeader createHeaderWithTimepointColumn() {
+    return new CsvHeader(
+        new String[] {
+          TRIP_ID_FIELD_NAME,
+          ARRIVAL_TIME_FIELD_NAME,
+          DEPARTURE_TIME_FIELD_NAME,
+          STOP_ID_FIELD_NAME,
+          STOP_SEQUENCE_FIELD_NAME,
+          STOP_HEADSIGN_FIELD_NAME,
+          PICKUP_TYPE_FIELD_NAME,
+          DROP_OFF_TYPE_FIELD_NAME,
+          CONTINUOUS_PICKUP_FIELD_NAME,
+          CONTINUOUS_DROP_OFF_FIELD_NAME,
+          SHAPE_DIST_TRAVELED_FIELD_NAME,
+          TIMEPOINT_FIELD_NAME
+        });
+  }
+
+  @Test
+  public void noTimepointColumn_noTimeProvided_shouldGenerateNotice() {
+    // unit test for legacy data that do not use timepoint column
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(null)
+            .setDepartureTime(null)
+            .setStopId("stop id 0")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(4)
+            .setTripId("second trip id")
+            .setArrivalTime(null)
+            .setDepartureTime(null)
+            .setStopId("stop id 1")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
+            .build());
+    assertThat(generateNotices(createLegacyHeader(), stopTimes))
+        .containsExactly(new MissingTimepointColumnNotice());
+  }
+
+  @Test
+  public void noTimepointColumn_timesProvided_shouldNotGenerateNotice() {
+    // unit test for legacy data that do not use timepoint column
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("second trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
```

Comment:
Is it right to set timepoint equal to `1` for this test? From the method name I expected both to be set to `null`.

Reply:
Follow up question after looking at other tests - how do you differentiate between "no column" and "no value" in the unit tests? 

Is the missing `timepoint`  value in `CsvHeader` header what triggers `MissingTimepointColumnNotice` (even if you call `.setTimepoint(1)` on one of the StopTime records)?

Is setting `.setTimepoint(null)` what is used to define a missing value (even if the `timepoint` value is included in header)?

I'd suggest clearly documenting this behavior in the unit tests as right now that's not clear (to me at least), even following the discussion thread in https://github.com/MobilityData/gtfs-validator/pull/1044/files#r742464479. And I'd avoid any confusing data that won't exist in the real world, like a missing column but a record with a timepoint value (i.e., this test currently).

Reply:
I removed this record in https://github.com/MobilityData/gtfs-validator/pull/1044/commits/e33c960a486db0db6d2af6a0b705fcfd6b44b026.

Reply:
> Follow up question after looking at other tests - how do you differentiate between "no column" and "no value" in the unit tests?

- no column -> these tests use he legacy header 
- no value -> these tests use the regular header, with `null` `timepoint` value. 

> Is the missing timepoint value in CsvHeader header what triggers MissingTimepointColumnNotice (even if you call .setTimepoint(1) on one of the StopTime records)?

Yes. 

> Is setting .setTimepoint(null) what is used to define a missing value (even if the timepoint value is included in header)?


Exactly. 

> I'd suggest clearly documenting this behavior in the unit tests as right now that's not 100% clear (to me at least), even following the discussion thread in https://github.com/MobilityData/gtfs-validator/pull/1044/files#r742464479. And I'd avoid any confusing data that won't exist in the real world, like a missing column but a record with a timepoint value (i.e., this test currently).

👍🏾 I removed the non-necessary (and misleading records) in https://github.com/MobilityData/gtfs-validator/commit/e33c960a486db0db6d2af6a0b705fcfd6b44b026. And added additional documentation in https://github.com/MobilityData/gtfs-validator/pull/1044/commits/84926f4eae651e30f0ac63f43c20a926056116fb.